### PR TITLE
docs: docstrings for parsers

### PR DESCRIPTION
**Pull request by cristianmtr**

### CODE REVIEW

File: jina/parsers/base.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+"""Module for """
```
Comment (alexcg1) [inline jina/parsers/base.py:1]: for...what?

File: jina/parsers/export_api.py [hunk 2]
```diff
@@ -1,3 +1,4 @@
+"""Argparser module for the export api"""
```
Comment (alexcg1) [inline jina/parsers/export_api.py:1]: API

File: jina/parsers/helper.py [hunk 3]
```diff
@@ -192,7 +201,11 @@ def _indents(self, line):
         return (indent, sub_indent)
 
     def _split_paragraphs(self, text):
-        """Split text in to paragraphs of like-indented lines."""
+        """Split text in to paragraphs of like-indented lines.
```
Comment (alexcg1) [inline jina/parsers/helper.py:204]: into

File: jina/parsers/peapods/runtimes/distributed.py [hunk 4]
```diff
@@ -1,15 +1,20 @@
+"""Argparser module for runtimes - distributed"""
```
Comment (alexcg1) [inline jina/parsers/peapods/runtimes/distributed.py:1]: for distributed runtimes (?)

This would be more consistent with below docstrings

**Review by alexcg1 — CHANGES_REQUESTED** [review]

**Review by alexcg1 — APPROVED** [review]
LGTM :+1:

File: jina/parsers/peapods/runtimes/distributed.py (hunk 4 above)
Comment (cristianmtr) [reply jina/parsers/peapods/runtimes/distributed.py]: Check again

File: jina/parsers/helper.py [hunk 5]
```diff
@@ -13,7 +15,11 @@ def add_arg_group(parser, title):
 
 
 def UUIDString(astring):
-    """argparse type to check if a string is a valid UUID string"""
+    """argparse type to check if a string is a valid UUID string
```
[the remark this answers is not in the document]
Comment (cristianmtr) [reply jina/parsers/helper.py:18]: Thanks. Check again

File: jina/parsers/ping.py [hunk 6]
```diff
@@ -1,3 +1,4 @@
+"""Argparser module for pinging"""
```
Comment (Yongxuanzhang) [inline jina/parsers/ping.py:1]: Miss doc for def set_ping_parser(parser=None).

File: jina/parsers/peapods/runtimes/remote.py [hunk 7]
```diff
@@ -1,3 +1,4 @@
+"""Argparser module for remote runtime"""
```
Comment (Yongxuanzhang) [inline jina/parsers/peapods/runtimes/remote.py:1]: Miss docs for methods in this file

File: jina/parsers/optimizer.py [hunk 8]
```diff
@@ -1,3 +1,4 @@
+"""Argparser module for Optimizer"""
```
Comment (Yongxuanzhang) [inline jina/parsers/optimizer.py:1]: Miss doc for set_optimizer_parser(parser: 'ArgumentParser' = None):

File: jina/parsers/hub/pushpull.py [hunk 9]
```diff
@@ -1,3 +1,4 @@
+"""Argparser module for hub push & pull"""
```
Comment (Yongxuanzhang) [inline jina/parsers/hub/pushpull.py:1]: Miss doc for mixin_hub_pushpull_parser(parser):

File: jina/parsers/hub/new.py [hunk 10]
```diff
@@ -1,3 +1,4 @@
+"""Argparser module for hub new"""
```
Comment (Yongxuanzhang) [inline jina/parsers/hub/new.py:1]: Miss doc for mixin_hub_new_parser(parser):

File: jina/parsers/hub/login.py [hunk 11]
```diff
@@ -1,3 +1,4 @@
+"""Argparser module for hub login"""
```
Comment (Yongxuanzhang) [inline jina/parsers/hub/login.py:1]: miss mixin_hub_docker_login_parser(parser):

File: jina/parsers/hub/list.py [hunk 12]
```diff
@@ -1,3 +1,4 @@
+"""Argparser module for hub list"""
```
Comment (Yongxuanzhang) [inline jina/parsers/hub/list.py:1]: miss mixin_hub_list_parser(parser):

File: jina/parsers/hub/build.py [hunk 13]
```diff
@@ -1,3 +1,4 @@
+"""Argparser module for hub build"""
```
Comment (Yongxuanzhang) [inline jina/parsers/hub/build.py:1]: miss mixin_hub_build_parser(parser):

File: jina/parsers/helper.py [hunk 14]
```diff
@@ -1,6 +1,8 @@
+"""Module for helper functions in the parser"""
```
Comment (Yongxuanzhang) [inline jina/parsers/helper.py:1]: miss doc for def add_arg_group(parser, title):

File: jina/parsers/helloworld.py [hunk 15]
```diff
@@ -1,3 +1,4 @@
+"""Module for hello world argparser"""
```
Comment (Yongxuanzhang) [inline jina/parsers/helloworld.py:1]: miss docs for funcs

File: jina/parsers/flow.py [hunk 16]
```diff
@@ -1,3 +1,4 @@
+"""Argparser module for Flow"""
```
Comment (Yongxuanzhang) [inline jina/parsers/flow.py:1]: miss def set_flow_parser(parser=None):

File: jina/parsers/export_api.py [hunk 17]
```diff
@@ -1,3 +1,4 @@
+"""Argparser module for the export API"""
```
Comment (Yongxuanzhang) [inline jina/parsers/export_api.py:1]: miss def set_export_api_parser(parser=None):

File: jina/parsers/client.py [hunk 18]
```diff
@@ -1,3 +1,4 @@
+"""Module for argparse for Client"""
```
Comment (Yongxuanzhang) [inline jina/parsers/client.py:1]: miss def mixin_client_cli_parser(parser):

File: jina/parsers/check.py [hunk 19]
```diff
@@ -1,3 +1,4 @@
+"""Argparser module for the check functions"""
```
Comment (Yongxuanzhang) [inline jina/parsers/check.py:1]: miss def set_check_parser(parser=None):

File: jina/parsers/base.py [hunk 20]
```diff
@@ -1,3 +1,4 @@
+"""Module containing the base parser for arguments of Jina."""
```
Comment (Yongxuanzhang) [inline jina/parsers/base.py:1]: miss def set_base_parser():

**Review by Yongxuanzhang — COMMENTED** [review]
Miss lots of docstrings for public methods. Not sure if we need to add? If so I can help to add in this PR or in another new PR. @cristianmtr

**Review by alexcg1 — DISMISSED** [review]

**Comment by cristianmtr** [comment]
> Miss lots of docstrings for public methods. Not sure if we need to add? If so I can help to add in this PR or in another new PR.
> Sorry I didn't find out these in the first review. Just use 'interrogate' to double-check the coverage
> @cristianmtr

Adding them now. I think I was running the wrong config of the linter here

File: jina/parsers/peapods/pod.py [hunk 21]
```diff
@@ -1,11 +1,14 @@
+"""Argparser module for Pod runtimes"""
 import argparse
 
 from jina.enums import PollingType, SchedulerType, PodRoleType
 from jina.parsers.helper import add_arg_group, _SHOW_ALL_ARGS
 
 
 def mixin_base_pod_parser(parser):
-    """Mixing in arguments required by :class:`BasePod` into the given parser. """
+    """Mixing in arguments required by :class:`BasePod` into the given parser.
```
Comment (nan-wang) [inline jina/parsers/peapods/pod.py:9]:
```suggestion
    """Add mixin arguments required by :class:`BasePod` into the given parser.
```

File: jina/parsers/__init__.py [hunk 22]
```diff
@@ -5,6 +5,12 @@
 
 
 def set_pea_parser(parser=None):
+    """
+    Set the parser for the Pea
```
Comment (nan-wang) [inline jina/parsers/__init__.py:9]: Let's stick to the same standard when writing docstrings. Previously, we are following the standard sphinx way, https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html#the-sphinx-docstring-format

```suggestion
    """Set the parser for the Pea
```

File: jina/parsers/__init__.py [hunk 23]
```diff
@@ -28,6 +34,12 @@ def set_pea_parser(parser=None):
 
 
 def set_pod_parser(parser=None):
+    """
+    Set the parser for the Pod
```
Comment (nan-wang) [inline jina/parsers/__init__.py:38]:
```suggestion
    """Set the parser for the Pod
```

File: jina/parsers/helper.py [hunk 24]
```diff
@@ -26,6 +39,15 @@ class KVAppendAction(argparse.Action):
     """
 
     def __call__(self, parser, args, values, option_string=None):
+        """
+        call the KVAppendAction
```
Comment (nan-wang) [inline jina/parsers/helper.py:43]:
```suggestion
        """Call :class:`KVAppendAction`
```

File: jina/parsers/hub/login.py [hunk 25]
```diff
@@ -1,9 +1,14 @@
...
 import os
 
 from ..helper import add_arg_group
 
 
 def mixin_hub_docker_login_parser(parser):
+    """
+    Add the options for the docker login
+    :param parser: the parser
```
Comment (nan-wang) [inline jina/parsers/hub/login.py:10]:
```suggestion
    """Add the options for the docker login
    
    :param parser: the parser
```

File: jina/parsers/hub/list.py [hunk 26]
```diff
@@ -1,7 +1,13 @@
...
 from ..helper import add_arg_group
 
 
 def mixin_hub_list_parser(parser):
+    """
+    Add the arguments for hub list to the parser
```
Comment (nan-wang) [inline jina/parsers/hub/list.py:7]:
```suggestion
    """Add the arguments for `hub list` to the parser
```

File: jina/parsers/hub/new.py [hunk 27]
```diff
@@ -1,7 +1,12 @@
...
 from ..helper import add_arg_group
 
 
 def mixin_hub_new_parser(parser):
+    """
+    Add the options for `hub new` to the parser
+    :param parser: the parser
+    """
```
Comment (nan-wang) [inline jina/parsers/hub/new.py:9]:
```suggestion
    """Add the options for `hub new` to the parser
    
    :param parser: the parser
    """
```

**Review by nan-wang — CHANGES_REQUESTED** [review]

File: jina/parsers/__init__.py (hunk 22 above)
Comment (cristianmtr) [reply jina/parsers/__init__.py:9]: The linter doesn't complain about it (as we agreed to configure it), so unless someone always checks them manually, I don't think we should prioritize this.

**Review by nan-wang — APPROVED** [review]
LGTM👍

**Pull request merged**